paywall/src/components/Paywall.js

```diff
@@ -98,7 +77,7 @@ export const mapStateToProps = ({ locks, keys, modals, router }) => {
 
   const modalShown = !!modals[locksFromUri.map(l => l.address).join('-')]
   const locked = validKeys.length === 0 || modalShown
-  return { locked, locks: locksFromUri, redirect }
+  return { locked, locks: locksFromUri, redirect, window }
```

Comment:
I would actually be arguing that having the `mapStateToProps` function pure (no global js object) is more important than having the react component themselves being pure, but again, I think we would need some research on this.